packages/core/src/internal/proc.js

```diff
@@ -40,21 +34,19 @@ function getIteratorMetaInfo(iterator, fn) {
   return getMetaInfo(fn)
 }
 
-// TODO: check if this hacky toString stuff is needed
-// also check again whats the difference between CHANNEL_END and CHANNEL_END_TYPE
+// toString is used to have ability to log actions in devtools.
 // maybe this could become MAYBE_END
 // I guess this gets exported so takeMaybe result can be checked
-export const CHANNEL_END = {
-  toString() {
-    return CHANNEL_END_SYMBOL
-  },
-}
+
 export const TASK_CANCEL = {
   toString() {
     return TASK_CANCEL_SYMBOL
   },
 }
 
+const isTerminated = res => res && res[TERMINATE]
```

Comment:
could u explain [this comment](https://github.com/redux-saga/redux-saga/pull/1464#discussion_r192986758)? Im not sure if I get what case are you talking about